### PR TITLE
simplify and fix sensitivites, add test

### LINKED ISSUE #1060: incorrect sensitivity mapping on trivial example
Urgent bug.

Here's a simple MWE:

```python
import numpy as np
import pandas as pd
from gpkit import Model, Variable, units


class Aircraft(Model):
    def setup(self):
        W_payload = Variable("W_{payload}", 175*(195 + 30), "lbf", "payload weight")
        f_oew = Variable("f_{oew}", 0.53, "-", "OEW/MTOW")
        fuel_per_nm = Variable("\\theta_{fuel}", 13.75, "lbf/nautical_mile", "fuel burn per nm")
        R = Variable("R", 3000, "nautical_miles", "range")
        mtow = Variable("MTOW", "lbf", "max take off weight")
        self.cost = 61.3e6*units.USD*(mtow/(1e5*units.lbf))**0.807
        return (mtow >= W_payload + f_oew*mtow + fuel_per_nm*R),
    
m = Aircraft()
sol = m.solve()
print sol.table()
print "sens to Wpay: %s" % sol["sensitivities"]["constants"]["W_{payload}"]
```

The result is (note sensitivity to W_payload should be non-zero):

```python
Using solver 'mosek'
Solving for 1 variables.
Solving took 0.0116 seconds.

Cost
----
 9.475e+07 [USD] 

Free Variables
--------------
MTOW : 1.715e+05  [lbf] max take off weight

Constants
---------
            R : 3000       [nmi]     range
  W_{payload} : 3.938e+04  [lbf]     payload weight
theta_{fuel} : 13.75      [lbf/nmi] fuel burn per nm
      f_{oew} : 0.53                 OEW/MTOW

Sensitivities
-------------
            R : +0.41 range
theta_{fuel} : +0.41 fuel burn per nm
      f_{oew} : +0.39 OEW/MTOW
  W_{payload} : +0    payload weight

sens to Wpay: 0.0
```

### PR DESCRIPTION
**Pull request by bqpd**
Closes #1060

@whoburg, ready for review.

### CODE REVIEW

**Comment by bqpd** [comment]
test models please

File: gpkit/nomials/data.py [hunk 1]
```diff
@@ -165,10 +165,11 @@ def simplify_exps_and_cs(exps, cs, return_map=False):
...
 
-    if len(matches) > 1:
-        zeroed_terms = (exp for exp, c in matches.items() if mag(c) == 0)
-        for exp in zeroed_terms:
-            del matches[exp]
+    zeroed_terms = (exp for exp, c in matches.items() if mag(c) == 0)
+    for exp in zeroed_terms:
+        del matches[exp]
+    if not matches:  # it's a zero monomial
+        matches[HashVector()] = 0
```
Comment (whoburg) [inline gpkit/nomials/data.py]: interesting, why do we need to set `matches[HashVector()] = 0` here? I would have thought we just let `exps_ = (),` and `cs = []` (and therefore `mmap = []` if returned) in that case.

File: gpkit/nomials/nomial_math.py [hunk 2]
```diff
@@ -454,34 +455,27 @@ def __init__(self, left, oper, right):
...
-        return tuple(exps_), cs/coeff
+            if not exp:
+                cs = cs.tolist()
+                coeff -= cs.pop(i)
+                cs = np.array(cs)/coeff
+                exps_ = list(exps)
+                exps_.pop(i)
+                exps = tuple(exps_)
+                if pmap is not None:
+                    self.const_mmap = pmap.pop(i)  # pylint: disable=attribute-defined-outside-init
```
Comment (whoburg) [inline gpkit/nomials/nomial_math.py]: whoa, this mutates pmap? That ought to be documented.

Comment (whoburg) [inline gpkit/nomials/nomial_math.py]: and is that really intended?

File: gpkit/nomials/nomial_math.py [hunk 3]
```diff
@@ -509,10 +503,10 @@ def as_posyslt1(self, substitutions=None):
         for posy in posys:
             _, exps, cs, subs = substitution(posy, substitutions)
             self._last_used_substitutions = subs
-            exps, cs = self._simplify_posy_ineq(exps, cs)
+            exps, cs, pmap = simplify_exps_and_cs(exps, cs, return_map=True)
+            exps, cs, pmap = self._simplify_posy_ineq(exps, cs, pmap)
```
Comment (whoburg) [inline gpkit/nomials/nomial_math.py]: There's a lot going on in the four lines above (substitution, and two simplification steps). Can you add an inline comment to clarify what the two different simplification steps are? Or perhaps make the method names more descriptive?

It's particularly interesting to me that the two simplification steps switched order in this PR -- what's up with that?

File: gpkit/nomials/data.py (hunk 1 above)
Comment (bqpd) [inline gpkit/nomials/data.py]: Not in the current scheme; the current abstraction for simplification is Bad, but this wasn't the PR to fix it...

File: gpkit/nomials/nomial_math.py (hunk 2 above)
Comment (bqpd) [inline gpkit/nomials/nomial_math.py]: Oh yes.

**Comment by bqpd** [comment]
Added comments per review.

**Pull request merged**